bootcd/rpms/genesis_scripts/genesis_scripts.spec

```diff
@@ -29,41 +30,55 @@ Scripts and configuration files used by Genesis in the bootcd image
 mkdir -p $RPM_BUILD_ROOT/root
 install -m 644 -T %{SOURCE0}   $RPM_BUILD_ROOT/root/.bash_profile.genesis_scripts
 
-# add some overrides we need
-mkdir -p $RPM_BUILD_ROOT/etc/sysconfig/network-scripts
-mkdir -p $RPM_BUILD_ROOT/etc/init
+# add our init scripts
 mkdir -p $RPM_BUILD_ROOT/etc/init.d
 install -m 755 -T %{SOURCE1}   $RPM_BUILD_ROOT/etc/init.d/network-prep
-install -m 644 -T %{SOURCE2}   $RPM_BUILD_ROOT/etc/sysconfig/init.diff
-install -m 644 -T %{SOURCE3}   $RPM_BUILD_ROOT/etc/init/tty.conf.override
+install -m 755 -T %{SOURCE6}   $RPM_BUILD_ROOT/etc/init.d/genesis
 
-# add helper for agetty
-install -m 555 -T %{SOURCE5}   $RPM_BUILD_ROOT/root/login-shell
+# add our config file
+mkdir -p $RPM_BUILD_ROOT/etc/sysconfig
+install -m 644 -T %{SOURCE8}   $RPM_BUILD_ROOT/etc/sysconfig/genesis
 
-# add the bootloader
+# add our binaries
 mkdir -p $RPM_BUILD_ROOT/usr/bin/
+# add helper for agetty
+install -m 555 -T %{SOURCE5}   $RPM_BUILD_ROOT/usr/bin/autologin
+# add the bootloader and its wrapper
 install -m 555 -T %{SOURCE4}   $RPM_BUILD_ROOT/usr/bin/genesis-bootloader
+install -m 755 -T %{SOURCE7}   $RPM_BUILD_ROOT/usr/bin/run-genesis-bootloader
 
 %clean
 # noop
 
 %files
 %defattr(-, root, root)
-%config /etc/init.d/network-prep
-%config /etc/sysconfig/init.diff
-%config /etc/init/tty.conf.override
-%config /root/.bash_profile.genesis_scripts
+/etc/init.d/genesis
+/etc/init.d/network-prep
+%config /etc/sysconfig/genesis
+/root/.bash_profile.genesis_scripts
 /usr/bin/genesis-bootloader
-/root/login-shell
+/usr/bin/run-genesis-bootloader
+/usr/bin/autologin
 
 %post
-cat /root/.bash_profile.genesis_scripts >> /root/.bash_profile
-# TODO undo this hack
-cp  /etc/init/tty.conf.override /etc/init/tty.conf
-/usr/bin/patch /etc/sysconfig/init < /etc/sysconfig/init.diff
+sed '/.bash_profile.genesis_scripts/d' -i /root/.bash_profile
```

Comment:
Scriptlets in rpm get the number of existing packages of this name on the system after the action completes. Use this to test in what context your scriptlet is executing and only perform profile inserts during new install 

From the fedora link below:

```
	install	upgrade	uninstall
 %pretrans	$1 == 0	$1 == 0	(N/A)
 %pre	$1 == 1	$1 == 2	(N/A)
 %post	$1 == 1	$1 == 2	(N/A)
 %preun	(N/A)	$1 == 1	$1 == 0
 %postun	(N/A)	$1 == 1	$1 == 0
 %posttrans	$1 == 0	$1 == 0	(N/A)

```

https://fedoraproject.org/wiki/Packaging:Scriptlets
https://www.redhat.com/archives/rpm-list/2002-August/msg00106.html